File: Java/프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Set<String>> map = new HashMap<>();
        Set<String> set;
        
        for (String[] cloth : clothes) {
            set = map.getOrDefault(cloth[1], new HashSet<>());
            set.add(cloth[0]);
            map.put(cloth[1], set);
        }
        
        for (String key : map.keySet()) {
            answer *= map.get(key).size() + 1;
        }
        
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.rb
def solution(clothes)
    answer = 1
    hash = { }
    
    clothes.each do |cloth|
        hash[cloth[1]] = 1 unless hash.has_key? cloth[1]
        hash[cloth[1]] += 1
    end
    
    hash.values.each { |v| answer *= v }
    
    return answer - 1
end